### DISCUSSION TITLE
Livewire 3.0 GET http://localhost/livewire/livewire.js?id=f41737f6 net::ERR_ABORTED 404 (Not Found)

### DISCUSSION THREAD

**Discussion by jxzsebastian** [livewire/livewire#6853] [1. Help]
I'm using laravel 10 with livewire 3.0  but i have a problem :(. when using the 'php artisan serve' command and go to "localhost:8000" does my project work without any issues, but when I navigate to the 'c:/xampp/htdocs/project/public' folder, Livewire doesn't work, and it shows the error mentioned in the title These are the configurations I have:

**config/livewire:**
return [

    'class_namespace' => 'App\\Livewire',

    'view_path' => resource_path('views/livewire'),

    'layout' => 'components.layouts.app',
    
    'lazy_placeholder' => null,

    'temporary_file_upload' => [
        'disk' => null,        // Example: 'local', 's3'              | Default: 'default'
        'rules' => null,       // Example: ['file', 'mimes:png,jpg']  | Default: ['required', 'file', 'max:12288'] (12MB)
        'directory' => null,   // Example: 'tmp'                      | Default: 'livewire-tmp'
        'middleware' => null,  // Example: 'throttle:5,1'             | Default: 'throttle:60,1'
        'preview_mimes' => [   // Supported file types for temporary pre-signed file URLs...
            'png', 'gif', 'bmp', 'svg', 'wav', 'mp4',
            'mov', 'avi', 'wmv', 'mp3', 'm4a',
            'jpg', 'jpeg', 'mpga', 'webp', 'wma',
        ],
        'max_upload_time' => 5, // Max duration (in minutes) before an upload is invalidated...
    ],

    'render_on_redirect' => false,

    'legacy_model_binding' => false,

    'inject_assets' => true,

    'navigate' => [
        'show_progress_bar' => true,
    ],

    'inject_morph_markers' => true,

    'pagination_theme' => 'tailwind',
];

**.env url:** 
APP_URL=http://localhost/AliadosFormulario/public

**and i changed this in apache/httpd.conf:**
<Directory />
    AllowOverride All
    Require all denied
</Directory>

**Comment by jrgeek20** [comment]
Hello, I have the same problem, right now I am trying to fix but I have not been able to find the issue.

**Reply by jxzsebastian** [reply]
I fixed the problem by changing the version of livewire to 2.12.6 and putting this in config/livewire:  'asset_url' => 'http://localhost/proyect/public',

**Comment by jrgeek20** [comment]
If you go to vendor/livewire/livewire/src/Mechanisms/FrontendAssets/FrontendAssets.php at line 157, you can fix this problem 
(http://localhost/livewire/livewire.js?id=f41737f6 net::ERR_ABORTED 404 (Not Found)): 

$url='http://localhost/mi-control/public/livewire/livewire.js';

But after to,  when you wanna play an event like "click" now you will got:

POST http://localhost/livewire/update 404 (Not Found)

=(

**Reply by GabrielCrespoBrito** [reply]
Same problem, fresh installation and nothing.

**Reply by jrgeek20** [reply]
Hello , the Solution:

https://livewire.laravel.com/docs/installation#publishing-config

grettings .

**Reply by GabrielCrespoBrito** [reply]
Thanks for the reply, unfortunately, it didn't work, same error. I have done a couple of tests, and it seems only when the Component uses the **mount** or **boot** method that it happens.

![Captura de pantalla 2023-09-28 172439](https://github.com/livewire/livewire/assets/94147533/f6cb7e80-063c-49c8-a457-b718b07924ca)

**Reply by jrgeek20** [reply]
Share a screnshot of the console log and the routes.config

**Reply by GabrielCrespoBrito** [reply]
The component works great when it does not have the **mount** method, but when you use this method, that path error appears, and it does not allow updates to the component to be made


![Captura de pantalla 2023-09-28 173812](https://github.com/livewire/livewire/assets/94147533/7fc79d8b-e6a3-42db-824e-c55f9ca0269b)
![Captura de pantalla 2023-09-28 173959](https://github.com/livewire/livewire/assets/94147533/07d337da-a91d-40ee-9e4d-0a510f59c210)

**Reply by jrgeek20** [reply]
As I suposed, first, if you want to learn the new Livewire version you really have to read quietly HAHAh!

I give you the solution in the first comment:

https://livewire.laravel.com/docs/installation#publishing-config

You have to add the next routes at the routes.conf:

Livewire::setUpdateRoute(function ($handle) {
    return Route::post('/yourcustomfolders/livewire/update', $handle);
});

  Livewire::setScriptRoute(function ($handle) {
    return Route::get('/yourcustomfolders/livewire/livewire.js', $handle);
});

Good luck.

**Reply by GabrielCrespoBrito** [reply]
Haha thanks trying to help me brother.

However, I still have the problem, I had already tried that solution.

As I told you, the update route works without any problem as long as the component does not have the **mount** method

I traced the code and it seems that

[$snapshot, $effects] = application('livewire')->update($snapshot, $updates, $calls);

The problem seems to occur, but as I said only when the component uses the mount method.

Anyway, thank you very much for your help.

` handleUpdate() function
  {
    $components = request('components');
    $responses = [];

    foreach ($components as $component) {
      $snapshot = json_decode($component['snapshot'], associative: true);
      $updates = $component['updates'];
      $calls = $component['calls'];

      -> [$snapshot, $effects] = application('livewire')->update($snapshot, $updates, $calls);
      

      $answers[] = [
        'snapshot' => json_encode($snapshot),
        'effects' => $effects,
      ];
    }

    $answer = [
      'components' => $responses,
    ];


    $finish = trigger('profile.response', $response);

    return $finish($response);
  }`

![image](https://github.com/livewire/livewire/assets/94147533/39b84e4c-0614-41a8-a72a-4426fd8b9346)

**Reply by jrgeek20** [reply]
Do you have AnyDesk ? i woul like revise your project. Send the anydesk access to jraulaggr@gmail.com

**Reply by GabrielCrespoBrito** [reply]
He just sent it to you by email.

**Reply by GabrielCrespoBrito** [reply]
For those who may be interested: it seems that the problem is in the hydration of the components, when I do not save model information in the components the error does not appear.

Credits to :

**Laraveloni** en https://laracasts.com/discuss/channels/livewire/livewire-3-throwing-404-after-dispatching-event

**Reply by Hunter-Slayer** [reply]
what if you dont have any custom but you still getting it when you clicked a button to do a certain function

**Comment by jrgeek20** [comment]
Try this:  
https://livewire.laravel.com/docs/installation#customizing-the-asset-url

**Comment by Mohammed-Thamnees** [comment]
I also has the same problem and i fixed it. Don't know whether its a right method or not. But i will share you.

After installing Liveware 3, just navigate to 'vendor/livewire/livewire'  there you can see a folder named 'dist' just copy the folder and paste it inside the Laravel public folder. After, rename the 'dist' folder inside the public folder to livewire. Inside the folder you can see the livewire.js file.  This worked for me...

**Reply by davitrix** [reply]
Tambien aplique esta misma solución, sin embargo, se supone que deberían de inyectarse automáticamente los assets, estoy usando laragon, en mi caso se inyectan correctamente los assets en local, pero no en Produccion

**Reply by focaicedoc** [reply]
@davitrix, basically, development and production are different context. By default, Livewire3 seek for their assets from public root URI of server. In Livewire2, you can set this via asset_url parameter in config/livewire.php configuration file. In Livewire3, you can't do it by default. Please, read the Livewire3 upgrade guide [here](https://livewire.laravel.com/docs/upgrading)

If your application is served under a non-root URI, you have two options to fix this:

1. set up you production server to serve your application from the public root URI of server
2. Customize you livewire asset_url following the standard procedure from documentation [here](https://livewire.laravel.com/docs/installation#customizing-the-asset-url)

Livewire3 have made a lot of things to you get more time to develop and create and  dedicate less time to configure it. But You can partially get back to Livewire2 style.

**Reply by davitrix** [reply]
Disculpe creo que falto contexto en mi comentario, estoy usando laragon como servidor, Laravel 10.x y livewire 3. 

Para el desarrollo local de mi proyecto uso la ruta raíz negobi.test con apache y en producción alojo mi proyecto en un vps negobi.com con nginx.

El problema es que cuando subo mi proyecto a producción el navegador no consigue la carpeta livewire/livewire que debería de inyectarse automáticamente en la carpeta public, sin embargo, esto en local si funciona correctamente.

En esta captura puede ver como se inyectan los activos de livewire correctamente en la carpeta public del proyecto trabajando en local, sin embargo, en producción esa carpeta nunca aparece y es la razon por la que copio los activos manualmente del directorio vendor a mi directorio public 
![Captura de pantalla 2024-03-02 074220](https://github.com/livewire/livewire/assets/43917766/4045b056-3394-48ab-83fc-c7697f27159d)

**Reply by AlyxsonMarques** [reply]
Man, thank you so much, a simple but REALLY fixing approach

**Reply by emrahyildiz1** [reply]
Worked perfectly 


After moving the dist folder to the public directory,  don't forget to write the redirect route in web.php

my web.php code 

Livewire::setScriptRoute(function ($handle) {
    return Route::get('/your-path/livewire.js', $handle);
});
Livewire::setUpdateRoute(function ($handle) {
    return Route::post('/your-path/livewire/update', $handle);
});

**Comment by focaicedoc** [comment]
Hello!

The address to fix this issue are detailed into [Livewire documentation](https://livewire.laravel.com/docs/installation#manually-bundling-livewire-and-alpine)

The directives

`import Clipboard from '@ryangjchandler/alpine-clipboard'`
`Alpine.plugin(Clipboard)`

are only necessary if you are using that plugin

**Comment by focaicedoc** [comment]
A permanent way to fix this issue:

1. check the APP_URL, in my case: APP_URL=http://localhost:8888/laravelapp/public
2. copy (or make a soft link) vendor/livewire/livewire/dist/livewire.js to public/build/assets/livewire.js
3. edit vendor/livewire/livewire/src/Mechanism/FrontendAssets/FrontendAssets.php boot function and replace

`return Route::get('/livewire/livewire.js', $handle);`

with
`return Route::get('/laravelapp/public/build/assets/livewire.js', $handle);`

4. clear routes, cache, views and config

I suspect livewire doesn't serve js assets correctly..., and we haven't a way to address this via configuration

**Comment by shubdragon** [comment]
Using Shared hosting or accessing through apache?

create a new `.htaccess` in root location of your project e.g the location where `.env` file resides .

and add this line to your .htaccess file after creating it if not exist (**Note: not talking about .htaccess present on public folder)

```
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteRule ^(.*)$ public/$1 [L]
</IfModule> 
``` 
this line internally rewrite the url and adds public to it.

This should solve your livewire update url not working or 404 issue

**Reply by erald93** [reply]
This worked for me, thanx

**Reply by domixinox** [reply]
Worked on shared hosting. Thanks

**Comment by khumam** [comment]
I got the same problem, perhaps it helps 
https://github.com/livewire/livewire/discussions/8085

**Comment by AbdelrhmanAmer** [comment]
1. Set the APP_URL in .env: Make sure your APP_URL in the .env file is correctly set to your local development URL. You can set it like this:

              `APP_URL=http://localhost`

2. Update the Livewire route: To fix the Livewire.js 404 error, add the following code to your routes/web.php file:


```
use Livewire\Livewire;

Livewire::setScriptRoute(function ($handle) {
    return Route::get('/laravel/myproject/vendor/livewire/livewire/dist/livewire.js', $handle);
});
```
3. Run the project: After making these changes, run your project using:
   
                  `php artisan serve`

**Comment by TUNBudi06** [comment]
<html>
<body>
<html><head></head><body>
<h1>🧩 Fix Livewire Deployment Error in Long Path (XAMPP or Subfolder)</h1>
<p>When deploying a <strong>Laravel Livewire</strong> project under a <strong>long or nested path</strong> (for example using <strong>XAMPP</strong> or in a subfolder),<br>
you may encounter this error:</p>
<pre><code>POST http://localhost/dev-app/public/livewire/update 404 (Not Found)
</code></pre>
<p>or sometimes it turns into something like:</p>
<pre><code>POST /laravel-bagan/public/laravel-bagan/public/livewire/update
</code></pre>
<p>This happens because <strong>Livewire automatically appends your <code inline="">APP_URL</code></strong> path twice when the project is inside a nested folder.</p>
<hr>
<h2>⚠️ Why This Happens</h2>
<p>Livewire dynamically generates internal script and update routes based on your <code inline="">APP_URL</code>.<br>
If your app lives inside a subfolder like:</p>
<pre><code>APP_URL=http://localhost/dev-app/public
</code></pre>
<p>Livewire will <strong>concatenate</strong> the path again when resolving its internal routes,<br>
causing the URL to double up like this:</p>
<pre><code>/dev-app/public/dev-app/public/livewire/update
</code></pre>
<hr>
<h2>✅ Easy Fix (XAMPP or Subfolder Deploy)</h2>
<p>You just need to <strong>manually define Livewire script and update routes</strong><br>
so that they match your real public path.</p>
<p>Follow these steps 👇</p>
<hr>
<h3>1. Publish Livewire Assets</h3>
<p>Run this command once:</p>
<pre><code class="language-bash">php artisan livewire:publish --assets
</code></pre>
<p>This will copy Livewire’s frontend assets into your public directory.</p>
<hr>
<h3>2. Define Custom Livewire Routes in <code inline="">AppServiceProvider</code></h3>
<p>Open your <code inline="">app/Providers/AppServiceProvider.php</code> and add this inside the <code inline="">register()</code> method:</p>
<pre><code class="language-php">use Livewire\Livewire;
use Illuminate\Support\Facades\Route;

class AppServiceProvider extends ServiceProvider
{
    /**
     * Register any application services.
     */
    public function register(): void
    {
        // Custom Livewire update route
        Livewire::setUpdateRoute(function ($handle) {
            return Route::post('/laravel-bagan/public/livewire/update', $handle);
        });

        // Custom Livewire script route
        Livewire::setScriptRoute(function ($handle) {
            return Route::get('/laravel-bagan/public/vendor/livewire/livewire.js', $handle);
        });
    }

    /**
     * Bootstrap any application services.
     */
    public function boot(): void
    {
        //
    }
}
</code></pre>
<blockquote>
<p>⚠️ Replace <code inline="">/laravel-bagan/public/</code> with your actual subfolder name if different (e.g. <code inline="">/dev-app/public/</code>).</p>
</blockquote>
<hr>
<h3>3. Register a Secondary Route (to Fix Double Path)</h3>
<p>Because Livewire may still append the subfolder again,<br>
you should <strong>redefine the route one more time</strong> in <code inline="">routes/web.php</code> to ensure both URLs are valid:</p>
<pre><code class="language-php">\Livewire\Livewire::setUpdateRoute(function ($handle) {
    return Route::post('/livewire/update', $handle);
});
</code></pre>
<p>Now Livewire will recognize both paths:</p>
<pre><code>/laravel-bagan/public/livewire/update
/livewire/update
</code></pre>
<hr>
<h2>🧠 Summary</h2>

Step | Description
-- | --
1️⃣ | Publish Livewire assets: php artisan livewire:publish --assets
2️⃣ | Register custom routes in AppServiceProvider
3️⃣ | Add fallback Livewire route in web.php
✅ | Test upload, form, or component actions — everything should now work under subfolder


<hr>
<h2>💡 Tips</h2>
<ul>
<li>
<p>If you later deploy to production (without subfolder), remove the hardcoded <code inline="">/public/</code> part to avoid duplicate routes.</p>
</li>
<li>
<p>You can also simplify by moving your Laravel app to root path (e.g. <code inline="">http://localhost/project/public</code>) for cleaner URLs.</p>
</li>
</ul>
</body>
</html>
